libraries/AP_GPS/AP_GPS.h

```diff
@@ -34,8 +34,6 @@
 // the number of GPS leap seconds
 #define GPS_LEAPSECONDS_MILLIS 18000ULL
 
-#define UNIX_OFFSET_MSEC (17000ULL * 86400ULL + 52 * 10 * MSEC_PER_WEEK - GPS_LEAPSECONDS_MILLIS)
```

Comment:
@OXINARF , @rmackay9.   I am sorry for your inconvenience ardupilot maintainer.
I will comment on the overflow.
The C language is matched to the largest type in the expression. The value of UNIX_OFFSET pointed out will be ULL.
Certainly, if "52 * 10 * MSEC_PER_WEEK" is the only one, UL becomes the largest type, so it will overflow.
＝＝＝
check program
#include <stdio.h>
#define GPS_LEAPSECONDS_MILLIS 18000ULL
#define MSEC_PER_SEC    1000UL
#define SEC_PER_WEEK    (7UL * 86400UL)
#define MSEC_PER_WEEK   (SEC_PER_WEEK * MSEC_PER_SEC)
#define UNIX_OFFSET_MSEC (17000ULL * 86400ULL + 52 * 10 * MSEC_PER_WEEK - GPS_LEAPSECONDS_MILLIS)

int main(int argc, char** argv)
{
    printf("%llx\n", 0ULL + 52 * 10 * MSEC_PER_WEEK);
    printf("%llX\n", UNIX_OFFSET_MSEC);
    return 0;
}
Execution result
(MSEC_PER_WEEK:49396c2000
UNIX_OFFSET_MSEC: 4990F7F5B0

===
@rmackay9 Please I would like to have a log file.